sites/fast-color-explorer/app/components/color-block.ts

```diff
@@ -0,0 +1,439 @@
+            :foregroundRecipe="${x => neutralForegroundRest}"
+        ></app-swatch>
+        <app-swatch
+            type="outline"
+            recipe-name="focusStrokeOuter"
+            :fillRecipe="${x => fillColor}"
+            :foregroundRecipe="${x => focusStrokeOuter}"
+            :outlineRecipe="${x => focusStrokeOuter}"
+        ></app-swatch>
+
+        <div class="example">
+            <fast-divider class="divider"></fast-divider>
+        </div>
+        <app-swatch
+            type="outline"
+            recipe-name="neutralStrokeDividerRest"
+            :fillRecipe="${x => fillColor}"
+            :foregroundRecipe="${x => neutralForegroundRest}"
+            :outlineRecipe="${x => neutralStrokeDividerRest}"
+        ></app-swatch>
+    </template>
+`;
+
+const template = html<AppColorBlock>`
+    <fast-design-system-provider
+        class="color-block"
+        fill-color="${x => x.backgroundColor}"
+        id="${x => x.backgroundColor.toUpperCase().replace("#", "")}"
+    >
+        <p class="title">
+            BACKGROUND ${x => x.index} - ${x => x.backgroundColor.toUpperCase()}
+            ${when(
+                x => x.layerName,
+                html`
+                    <p>
+                        <code>${x => x.layerName}</code>
+                    </p>
+                `
+            )}
+        </p>
+
+        <div class="content">
+            ${when(x => x.component === ComponentTypes.backplate, backplateComponents)}
+            ${when(x => x.component === ComponentTypes.text, textComponents)}
+            ${when(x => x.component === ComponentTypes.form, formComponents)}
+        </div>
+    </fast-design-system-provider>
+`;
+
+const styles = css`
+    :host {
+        min-height: 100%;
+    }
+
+    .color-block {
+        display: flex;
+        flex-direction: column;
+        flex-grow: 1;
+        align-items: stretch;
+        text-align: center;
+        color: ${neutralForegroundRest};
+        position: relative;
+        transition: opacity 0.1s linear;
+        height: 100%;
+    }
+
+    .title {
+        margin: 16px auto 4px;
+        font-weight: 600;
+        height: 34px;
+        color: ${neutralForegroundHint};
+    }
+
+    .title code {
+        font-weight: normal;
+    }
+
+    .content {
+        flex-grow: 1;
+        display: flex;
+        flex-direction: column;
+        align-items: center;
+        padding: 0 48px 36px;
+    }
+
+    .example {
+        height: 60px;
+        display: flex;
+        align-items: center;
+        margin-top: 24px;
+    }
+
+    .hint_text {
+        color: ${neutralForegroundHint};
+    }
+
+    .divider {
+        width: 150px;
+    }
+`;
+
+@customElement({
+    name: "app-color-block",
+    template,
+    styles,
+})
+export class AppColorBlock extends FASTElement {
+    @attr index: number;
+
+    @attr component: ComponentTypes;
+
+    @attr({ attribute: "background-color" }) backgroundColor: string;
```

Comment:
should this be `fill-color`?

Reply:
It does apply to fill-color, but I was keeping it separate from that. I went with 'color' since this is the 'color-block'.